src/Ormolu/Printer/Meat/Common.hs

```diff
@@ -89,6 +90,25 @@ p_ieWildcard = \case
   NoIEWildcard -> return ()
   IEWildcard n -> parens (atom n)
 
+-- | A helper for pretty printing groups of certain syntactical elements.
+-- Groups are printed without any newlines in-between, whereas separate groups
+-- have an additional newline for clarity.
+
+p_groupingHelper
+  :: (a -> R ())                -- ^ How to format elements
+  -> (a -> a -> Bool)           -- ^ Whether two declarations should be grouped
+  -> [a]                        -- ^ List of syntactic elements
+  -> R ()
+p_groupingHelper pretty group xs =
```

Comment:
It seems that `group` is a misnomer here, it is actually a predicate that says if they should be separated!